solvebio/query.py

```diff
@@ -412,12 +431,20 @@ def next(self):
             # Iterator not initialized yet
             self.__iter__()
 
+        # check if a current object is the join query
+        _is_join = getattr(self, '_is_join', False)
+
         # len(self) returns `min(limit, total)` results
-        if self._cursor == len(self):
+        if not _is_join and self._cursor == len(self):
             raise StopIteration
 
         if self._buffer_idx == len(self._buffer):
-            self.execute(self._page_offset + self._buffer_idx)
+            if _is_join:
+                if self._next_offset >= self._limit:
```

Comment:
Add a comment:

# Since joins can return more results than we expect (due to `explode`)
# manually ensure that we haven't gone above the requested limit (default inf)